### PR TITLE
[Minor] Improve readability by replacing lambda with method reference

### LINKED ISSUE #487: [Improvement] Java 8 syntax helps with code readability improvements
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Can be replaced with the lambda referenced by the method. Method references are easier to understand and read and unify than lambda.

### How should we improve?

I will submit the changed PR

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by iwangjie**
### What changes were proposed in this pull request?

Improve readability by replacing lambda with method reference.

### Why are the changes needed?

Fixes #487

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing CI.

### CODE REVIEW

File: coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public SimpleClusterManager(CoordinatorConf conf, Configuration hadoopConf) thro
 
     periodicOutputIntervalTimes = conf.get(CoordinatorConf.COORDINATOR_NODES_PERIODIC_OUTPUT_INTERVAL_TIMES);
     scheduledExecutorService.scheduleAtFixedRate(
-        () -> nodesCheck(), heartbeatTimeout / 3,
+            this::nodesCheck, heartbeatTimeout / 3,
```
Comment (kaijchen) [inline coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java:86]: Please don't change the indentation here.

**Review by kaijchen — COMMENTED** [review]
Thanks for your contribution @iwangjie. Mostly LGTM, but there are some style issues.
Also, please fill in the pull request template as much as possible.

File: server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java [hunk 2]
```diff
@@ -116,17 +116,17 @@ public ShuffleTaskManager(
     this.scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
         ThreadUtils.getThreadFactory("checkResource-%d"));
     scheduledExecutorService.scheduleAtFixedRate(
-        () -> preAllocatedBufferCheck(), preAllocationExpired / 2,
+            this::preAllocatedBufferCheck, preAllocationExpired / 2,
```
Comment (kaijchen) [inline server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:119]: @iwangjie there are a few more cases in this file.

File: server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java [hunk 3]
```diff
@@ -116,17 +116,17 @@ public ShuffleTaskManager(
...
         ThreadUtils.getThreadFactory("checkResource-%d"));
     scheduledExecutorService.scheduleAtFixedRate(
-        () -> preAllocatedBufferCheck(), preAllocationExpired / 2,
+        this::preAllocatedBufferCheck, preAllocationExpired / 2,
         preAllocationExpired / 2, TimeUnit.MILLISECONDS);
     this.expiredAppCleanupExecutorService = Executors.newSingleThreadScheduledExecutor(
         ThreadUtils.getThreadFactory("expiredAppCleaner"));
     expiredAppCleanupExecutorService.scheduleAtFixedRate(
-        () -> checkResourceStatus(), appExpiredWithoutHB / 2,
+            this::checkResourceStatus, appExpiredWithoutHB / 2,
```
Comment (kaijchen) [inline server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:124]:
```suggestion
        this::checkResourceStatus, appExpiredWithoutHB / 2,
```

File: server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java [hunk 4]
```diff
@@ -116,17 +116,17 @@ public ShuffleTaskManager(
...
         appExpiredWithoutHB / 2, TimeUnit.MILLISECONDS);
     this.leakShuffleDataCheckExecutorService = Executors.newSingleThreadScheduledExecutor(
         ThreadUtils.getThreadFactory("leakShuffleDataChecker"));
     leakShuffleDataCheckExecutorService.scheduleAtFixedRate(
-        () -> checkLeakShuffleData(), leakShuffleDataCheckInterval,
+            this::checkLeakShuffleData, leakShuffleDataCheckInterval,
```
Comment (kaijchen) [inline server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java:129]:
```suggestion
        this::checkLeakShuffleData, leakShuffleDataCheckInterval,
```

**Review by kaijchen — CHANGES_REQUESTED** [review]
Please take a glance of your changes in the [Files changed](https://github.com/apache/incubator-uniffle/pull/488/files) tab in the PR page.

**Comment by iwangjie** [comment]
> Please take a glance of your changes in the [Files changed](https://github.com/apache/incubator-uniffle/pull/488/files) tab in the PR page.

🥲 Thank you. I will avoid problems with code style checking and contribution specifications. sincere greetings.

**Comment by kaijchen** [comment]
> 🥲 Thank you. I will avoid problems with code style checking and contribution specifications. sincere greetings.

It's OK. Everybody has to start somewhere. Thanks for updating the PR.

**Review by kaijchen — APPROVED** [review]
LGTM, pending CI.

**Comment by kaijchen** [comment]
@iwangjie, congrats on your first commit on Uniffle.

**Pull request merged**